components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -486,4 +687,141 @@ private SessionManagementClientException handleSessionManagementClientException(
         }
         return new SessionManagementClientException(error, description);
     }
+
+    /**
+     * This method checks whether federated associations exist for the given userId and if so returns the
+     * internal userId stored in IDN_AUTH_USER table.
+     *
+     * @param userId User Id.
+     * @param tenantId Tenant Id.
+     * @return User id to which sessions are stored against.
+     */
+    private String getUserIdFromFederatedMapping(int tenantId, String userId) throws UserSessionException {
```

Comment:
```suggestion
    private String getAuthSessionUserIdFromAssociatedMapping(int tenantId, String userId) throws UserSessionException {
```
Think on a better name